src/services/api.tsx

```diff
@@ -39,7 +44,109 @@ export const getPatients = async () => {
         if (name[0].given) patient.firstName = name[0].given.join(", ");
         if (name[0].family) patient.lastName = name[0].family;
       }
+      if (identifier) {
+        patient.identifier = identifier
+          .map((e: any) => {
+            return e.value;
+          })
+          .join(", ");
+      }
+
       return patient;
     }
   );
 };
+
+export const getPatientData = (patientId: string) => {
+  /*
+    getPatientData requests for data from Patient resourcce of id patientId
+    return a Patient object.
+  */
+  const getPatientData = async () => {
+    let response = await client.search({
+      type: "Patient",
+      patient: patientId,
+      query: {}
+    });
+    if (!response.data.entry) return; //patient not found
+    const patientData = response.data.entry[0];
+
+    const patient: Patient = {
+      id: patientData.resource.id
+    };
+
+    // Completing patient information with available data
+    if (patientData.resource.identifier) {
+      patient.identifier = patientData.resource.identifier[0];
+    }
+
+    if (patientData.resource.birthDate) {
+      patient.age = getAge(new Date(patientData.resource.birthDate));
+      patient.birthDate =
+        patientData.resource.birthDate +
+        " (" +
+        patient.age.toString() +
+        " ans)";
+    }
+    if (patientData.resource.name) {
+      if (patientData.resource.name[0].given)
+        patient.firstName = patientData.resource.name[0].given.join(", ");
+      if (patientData.resource.name[0].family)
+        patient.lastName = patientData.resource.name[0].family;
+    }
+
+    if (patientData.resource.identifier) {
+      patient.identifier = patientData.resource.identifier
+        .map((e: any) => {
+          return e.value;
+        })
+        .join(", ");
+    }
+
+    const responseAI = await getPatientResources(
+      "AllergyIntolerance",
+      patientId
+    );
+    const responseO = await getSubjectResources("Observation", patientId);
+    const responseC = await getSubjectResources("Condition", patientId);
+    const responseEoC = await getPatientResources("EpisodeOfCare", patientId);
+
+    patient.number = {
```

Comment:
rename to something more meaningful (as discussed)

Reply:
Modified